filebeat/input/filestream/internal/input-logfile/clean.go

```diff
@@ -130,5 +130,5 @@ func checkCleanResource(started, now time.Time, resource *resource) bool {
 		reference = started
 	}
 
-	return reference.Add(ttl).Before(now) && resource.stored
+	return (ttl >= 0 && reference.Add(ttl).Before(now)) && resource.stored
```

Comment:
Please add a comment here too, something like "If ttl is negative we never delete it, otherwise check how much time is elapsed."

Nit: the extra parentheses aren't needed since it's just an && of three conditions. Optionally you could also do an immediate `if ttl < 0 { return }` above so the extra clause isn't needed here.